backend/api/Services/ReportService.cs

```diff
@@ -4,9 +4,38 @@
 
 namespace Api.Services
 {
+    public interface IReportService
+    {
+        public abstract Task<Report> Create(Report report);
+
+        public abstract Task<Report> Create(
+            string isarMissionId,
+            string echoMissionId,
+            string log,
+            ReportStatus status,
+            Robot robot
+        );
+
+        public abstract Task<IEnumerable<Report>> ReadAll();
+
+        public abstract Task<Report?> Read(string id);
+
+        public abstract Task<Report?> ReadByIsarMissionId(string isarMissionId);
+
+        public abstract Task<IsarTask?> ReadIsarTaskById(string isarTaskId);
+
+        public abstract Task<IsarStep?> ReadIsarStepById(string isarStepId);
+
+        public abstract Task<bool> UpdateMissionStatus(string isarMissionId, ReportStatus reportStatus);
+
+        public abstract Task<bool> UpdateTaskStatus(string isarTaskId, IsarTaskStatus taskStatus);
+
+        public abstract Task<bool> UpdateStepStatus(string isarStepId, IsarStep.IsarStepStatus stepStatus);
+    }
+
```

Comment:
I could use an explanation on this programming pattern.
Are we planning to use several implementations of this interface?

Reply:
Yes, sure!
The use interfaces for services is recommended for scalability (In which case they should be in their own files, of course), they also serve as a good overview/documentation of the intended key features of a service.
In this case I implemented them because they greatly reduce the need for mocking of code in the test project, which is a nice bonus.